### PR TITLE
devcontainer: network connectivity

### PR DESCRIPTION
**Pull request by ne20002**
Updated configuration mainly for network settings of the dev container.
- friendica instance in the container can now be reached on port 80, 8080 and 443
- create self-signed certificate upon creation of the dev container
- updated configuration to make use of values defined in .env file

### CODE REVIEW

**Comment by ne20002** [comment]
Hi @friendica 

As you may have seen I'm trying to set up a dev container that can be used with vscode or other tools supporting dev containers (it's not only vscode).
The current setup (with the additions of this pr) is working quite well even though a few things are still missing or not working (e.g. tests). I followed what I found in the Vagrant files in this repository.

I now found that it should be possible to, instead of creating a dev container fully and from scratch, build the dev container by using ddev inside the dev container.

I'm unsure about how you have set up your development environments so I'm asking for your vote:
- continue creating a new independent dev container from scratch
- use ddev inside the devcontainer which would keep it in sync with a ddev setup in the repo
- build the dev container based on something else?

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
 	// "features": {},
 
 	// For use with PHP or Apache (e.g.php -S localhost:8080 or apache2ctl start)
-	"forwardPorts": [3306, 8080],
+	//"forwardPorts": [3306, 8080],
```
Comment (MrPetovan) [inline .devcontainer/devcontainer.json]: May this commented out line need to be uncommented at some point? Otherwise, please remove it.

**Comment by MrPetovan** [comment]
@annando develops in production, while I set up a second node to work on. I fully trust you about the creation of this devcontainer because I've never needed the feature.

**Comment by annando** [comment]
@ne20002 yeah, like @MrPetovan already said, my Friendica node is a production server and development server in one. And yeah, I know that this not the recommended way, but it works fine for me and my users are fine with it as well.

**Comment by ne20002** [comment]
I like the concept of the development container. I prevents me from bloating my machine with lots of software as this is what usually happens when you try to work on different projects with different environments. 

@MrPetovan if you set up a second node how did you do this? Using ddev?

Looking in the repo I found files to help setting up the dev environment with Vagrant or with DDev. But I don't know if this is still used or outdated. 
As far as I understand, DDev is a setup to make it easier to develop php by putting some of the needed additional functionalities (web-server, database, ...) into containers but the development is still done on a machine by plain access. 
A dev container now moves the full develoment environment with all needed software into a container that can be build and set up with a single command from a repository. (And you can even run that completely in a GitHub codespace). It makes it also very easy to update the underlying os and development software.

I believe developers already working on the project have their own setup and are happy with it. I want to create a development container to make it easier for new contributers. And also to improve my Docker knowledge.

If there are friendica developers using the DDev setup it might be a good idea to build a dev container which includes a DDev setup so this would be in sync (and easier to maintain in the future). That's why I ask.

At the moment the dev container can spin up an environment similar to the Vagrant setup with a working empty instance and developing including debugging works. Tests are not working, I still try to understand how this test database works. And a few additional steps as database backup, export and import or triggering cron is still missing.

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (ne20002) [reply .devcontainer/devcontainer.json]: Done

**Comment by annando** [comment]
@ne20002 are you done with your changes?

**Comment by ne20002** [comment]
Yes. I decided against ddev container (didn't get it running satisfactorily). So: this is the way.

**Pull request merged**